addons/im_livechat/models/discuss_channel.py

```diff
@@ -289,6 +300,13 @@ def _message_post_after_hook(self, message, msg_vals):
                 }
             )
 
+        if (
+            self.livechat_active and
+            message.author_id == self.livechat_operator_id and
+            message.author_id != self.chatbot_current_step_id.chatbot_script_id.operator_partner_id
+        ):
```

Comment:
`livechat_operator_id` should be considered somewhat deprecated, use member history table to know who's agent
then checking bot is not necessary either